src/vkvg_context.c

```diff
@@ -112,6 +112,7 @@ VkvgContext vkvg_create(VkvgSurface surf)
 	}
 
 	ctx->pSurf = surf;
+	vkvg_surface_reference (surf);
```

Comment:
This is not the right place to put it. It must come after surface state test! Also what if the context is reused? Have you noticed that there are 2 pathes for initializing context. An existing context cached in the device could be reused, surface reference as also to be done there..

Reply:
Right, what about this https://github.com/Rubo3/vkvg/commit/a0c5684b9d56c54d0063a10f312db8ead498db39?